File: lib/model/product.dart
class Product {
  int pk;
  String serial;
  String name;
  String type;
  String state;
  int available;

  Product({required this.pk,
    required this.serial,
    required this.name,
    required this.type,
    required this.state,
    required this.available,
    }) {}

  factory Product.def(){
    return Product(
        pk: -1,
        serial: '',
        name: '',
        type: '',
        state: "N/A",
        available: 0,
        );
  }

  factory Product.fromJson(Map<String, dynamic> Json) {
    return Product(
        pk: Json['pkProd'] as int,
        serial: Json['serialProd'],
        name: Json['nameProd'],
        type: Json['typeProd'],
        state: Json['stateProd'],
        available: Json['available'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'pkProd': this.pk,
      'serialProd': this.serial,
      'nameProd': this.name,
      'typeProd': this.type,
      'stateProd': this.state,
      'available': this.available,
    };
  }
}

File: server/src/routes/product.js
import Router from 'express'
import db from '../db/connection.js'
const router = Router();

const query_product = "SELECT * FROM Products WHERE " +
  "Products.serialProd = ?;";
const ins_product = "INSERT INTO Products (serialProd, nameProd, " +
  "typeProd, stateProd, available) " +
  "VALUES " +
  "(?, ?, ?, ?, ?);";

router.get('/products/:serial', (req, res) => {
  db.query(query_product, [req.params.serial], (err, rows, field) => {
    if (err) throw err;
    res.json(rows[0]);
    console.log(rows[0]);
  });
});

router.post('/products', (req, res) => {
  console.log("products post access");
  db.query(ins_product, [
    req.body.serialProd,
    req.body.nameProd,
    req.body.typeProd,
    req.body.stateProd,
    req.body.available
  ], (err, result) => {
      if (err) throw err;
      console.log(result);
    })
});


export default router;
